File: go/go_teste/calculadora.go
package main

import "fmt"

func main() {
    var num1, num2 float64
    var operator string

    fmt.Println("Calculadora em Go")
    fmt.Println("-----------------")

    // Solicitação para inserir o primeiro número
    fmt.Print("Digite o primeiro número: ")
    fmt.Scanln(&num1)

    // Solicitação para inserir o operador (+, -, *, /)
    fmt.Print("Digite o operador (+, -, *, /): ")
    fmt.Scanln(&operator)

    // Solicitação para inserir o segundo número
    fmt.Print("Digite o segundo número: ")
    fmt.Scanln(&num2)

    var resultado float64

    // Realiza a operação com base no operador fornecido
    switch operator {
    case "+":
        resultado = num1 + num2
    case "-":
        resultado = num1 - num2
    case "*":
        resultado = num1 * num2
    case "/":
        if num2 != 0 {
            resultado = num1 / num2
        } else {
            fmt.Println("Erro: divisão por zero")
            return
        }
    default:
        fmt.Println("Operador inválido")
        return
    }

    // Exibe o resultado da operação
    fmt.Printf("%.2f %s %.2f = %.2f\n", num1, operator, num2, resultado)
}

File: c++/calculadora.cpp
#include <iostream>

using namespace std;

void menu() {
    cout << "Escolha uma operação:" << endl;
    cout << "1. Adição" << endl;
    cout << "2. Subtração" << endl;
    cout << "3. Multiplicação" << endl;
    cout << "4. Divisão" << endl;
    cout << "5. Sair" << endl;
}

int main() {
    int escolha;
    double num1, num2;
    bool continuar = true;

    while (continuar) {
        menu();
        cout << "Digite a sua escolha (1-5): ";
        cin >> escolha;

        if (escolha == 5) {
            continuar = false;
            cout << "Saindo da calculadora..." << endl;
            continue;
        }

        if (escolha < 1 || escolha > 5) {
            cout << "Escolha inválida. Tente novamente." << endl;
            continue;
        }

        cout << "Digite o primeiro número: ";
        cin >> num1;
        cout << "Digite o segundo número: ";
        cin >> num2;

        switch (escolha) {
            case 1:
                cout << "Resultado: " << num1 + num2 << endl;
                break;
            case 2:
                cout << "Resultado: " << num1 - num2 << endl;
                break;
            case 3:
                cout << "Resultado: " << num1 * num2 << endl;
                break;
            case 4:
                if (num2 != 0) {
                    cout << "Resultado: " << num1 / num2 << endl;
                } else {
                    cout << "Erro: Divisão por zero!" << endl;
                }
                break;
        }
    }

    return 0;
}
